src/pineko/utils.py

```diff
@@ -29,14 +29,17 @@ def read_grids_from_nnpdf(dataset_name, configs=None):
         return None
 
     # Import NNPDF only if we really want it!
-    from nnpdf_data import legacy_to_new_map
-    from validphys.commondataparser import EXT
-    from validphys.loader import Loader
+    from nnpdf_data import legacy_to_new_map, path_commondata
+    from nnpdf_data.commondataparser import EXT, parse_new_metadata
 
-    # We only need the metadata, so this should be enough
+    # We only need the metadata, so this should be enough.
+    # Pass it through the legacy_to_new in case this is an old name.
     dataset_name, variant = legacy_to_new_map(dataset_name)
-    cd = Loader().check_commondata(dataset_name, variant=variant)
-    fks = cd.metadata.theory.FK_tables
+
+    setname, observable = dataset_name.rsplit("_", 1)
+    metadata_file = path_commondata / setname / "metadata.yaml"
+    metadata = parse_new_metadata(metadata_file, observable, variant=variant)
+    fks = metadata.theory.FK_tables
```

Comment:
it's fine to have this stuff here for the moment, but shouldn't that all be moved eventually up into `nnpdf-data`? all that information is very specific to that package

Reply:
actually, the last line (`fks = ...`) will be left here

Reply:
You're probably right, but putting it in nnpdf-data would result in some double counting with the regular nnpdf package which is also not ideal

Reply:
To be looked at after this is actually used.